tools/alara_output_processing.py

```diff
@@ -0,0 +1,399 @@
+                results (dict): Dictionary that stores all parsed tables,
+                    keyed by parameter and block name.
+                current_parameter (str): Specific quantitative value
+                    represented in the table (e.g. specific activity, number
+                    density, etc.)
+                current_interval (str): Interval iterated upon in ALARA run.
+
+            Returns:
+                None
+            '''
+
+            df = pd.read_csv(
+                StringIO('\n'.join(current_table_lines)),
+                sep=r'\s+'
+            )
+
+            df.columns = [c.replace('_', '') for c in df.columns]
+            key = f'{current_parameter} - {current_block}'
+            self.results[key] = df
+
+    def parse_output(self):
+        '''
+        Reads an ALARA output file, identifies all data tables contained
+            within, and stores each as a Pandas DataFrame in a dictionary.
+
+        Arguments:
+            self
+
+        Returns:
+            results (dict): Dictionary that stores all parsed tables,
+                keyed by parameter and block name.
+        '''
+
+        with open(self.filepath, 'r') as f:
+            lines = f.readlines()
+
+        current_parameter = None
+        current_block = None
+        inside_table = False
+        current_table_lines = []
+
+        for line in lines:
+            line = line.strip()
+
+            if self.is_new_parameter(line):
+                current_parameter = line.strip('* ').strip()
+                continue
+
+            if self.is_new_block(line):
+                current_block = line.rstrip(':')
+                continue
+
+            if self.is_table_header(line):
+                inside_table = True
+                current_table_lines = [self.normalize_header(line)]
+                continue
+
+            if inside_table and self.is_separator(line):
+                continue
+
+            if inside_table:
+                current_table_lines.append(line)
+                if self.is_end_of_table(line):
+                    if current_parameter and current_block:
+                        self._parse_table_data(
+                            current_table_lines,
+                            current_parameter,
+                            current_block
+                        )
+                    inside_table = False
+                    current_table_lines = []
+                continue
+
+        return self.results
+    
+    # ---------- Output ----------
+    def write_csv_files(self):
+        '''
+        Write out all DataFrames extracted from parsed ALARA output tables to
+            their own CSV files.
+
+        Arguments:
+            self
+        
+        Returns:
+            None
+        '''
+
+        for key, df in self.results.items():
+            filename = self.sanitize_filename(key) + '.csv'
+            df.to_csv(filename, index=False)
+
+class DataProcessing:
```

Comment:
If we separate out the `ALARADFrame` to its own class, then maybe this one can be called `DataLibrary`?